### PR TITLE
ARXIVNG-413 docker-compose provides a working demonstration

### PR DESCRIPTION
**Pull request by erickpeirson**
Depends on #18.

From ``README.md``:

## Local deployment (for testing only)

A Compose file ([docker-compose.yml](docker-compose.yml)) is included in the
root of this repository, and can be used to run the services in this project
for local testing and development purposes.

See the [Docker documentation](https://docs.docker.com/compose/) for
information about using Docker Compose.

The Compose file included here deploys all services on a custom network, and
exposes the gateway service at port 8000 on your local machine.

To start up (from the root of the repo, containing ``docker-compose.yml``):

```bash
$ docker-compose build
$ docker-compose up
```

This will generate a lot of output, and bind your terminal. Some things to
watch for:

MariaDB starting up successfully:
```
submission-maria            | 2018-04-11 13:49:16 0 [Note] Reading of all Master_info entries succeded
submission-maria            | 2018-04-11 13:49:16 0 [Note] Added new Master_info '' to hash table
submission-maria            | 2018-04-11 13:49:16 0 [Note] mysqld: ready for connections.
submission-maria            | Version: '10.3.5-MariaDB-10.3.5+maria~jessie'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
```

The metadata API service waiting for the database to be ready:

```
submission-metadata         | application 11/Apr/2018:09:49:13 +0000 - __main__ - None - [arxiv:null] - INFO: "...waiting 4 seconds..."
```

After the DB is available, you should see something like this as the metadata
service initializes the database and adds some data.

```
submission-metadata         | application 11/Apr/2018:09:49:17 +0000 - __main__ - None - [arxiv:null] - INFO: "Checking for database"
submission-metadata         | application 11/Apr/2018:09:49:17 +0000 - __main__ - None - [arxiv:null] - INFO: "Database not yet initialized; creating tables"
submission-metadata         | application 11/Apr/2018:09:49:18 +0000 - __main__ - None - [arxiv:null] - INFO: "Populate with base data..."
submission-metadata         | application 11/Apr/2018:09:49:18 +0000 - __main__ - None - [arxiv:null] - INFO: "Added 10 licenses"
submission-metadata         | application 11/Apr/2018:09:49:18 +0000 - __main__ - None - [arxiv:null] - INFO: "Added 3 policy classes"
submission-metadata         | application 11/Apr/2018:09:49:18 +0000 - __main__ - None - [arxiv:null] - INFO: "Added 150 categories"
submission-metadata         | application 11/Apr/2018:09:49:18 +0000 - __main__ - None - [arxiv:null] - INFO: "Added 500 users"
```

Note that the users generated here are fake. Licenses, categories, etc are all
"realistic".

After generating data, the web service starts:

```
submission-metadata         | spawned uWSGI master process (pid: 13)
submission-metadata         | spawned uWSGI worker 1 (pid: 21, cores: 100)
submission-metadata         | spawned uWSGI worker 2 (pid: 22, cores: 100)
submission-metadata         | spawned uWSGI worker 3 (pid: 23, cores: 100)
submission-metadata         | spawned uWSGI worker 4 (pid: 24, cores: 100)
submission-metadata         | spawned uWSGI worker 5 (pid: 25, cores: 100)
submission-metadata         | spawned uWSGI worker 6 (pid: 26, cores: 100)
submission-metadata         | spawned uWSGI worker 7 (pid: 27, cores: 100)
submission-metadata         | spawned uWSGI worker 8 (pid: 28, cores: 100)
```

At this point, you should be able to interact with the submission API. E.g.

```bash
$ curl -i -X POST -H "Content-Type: application/json" -H "Authorization: Bearer as392lks0kk32" --data-binary "@metadata/examples/complete_submission.json" http://localhost:8000/submission/
```

### CODE REVIEW

File: Dockerfile-metadata [hunk 1]
```diff
@@ -2,26 +2,29 @@
 
 FROM arxiv/base
 
-RUN pip install uwsgi
+WORKDIR /opt/arxiv
+
+# We're using MariaDB for testing/development, but production is MySQL (for
```
Comment (mhl10) [inline Dockerfile-metadata]: Just a caution that while MariaDB 10.3.x and MySQL 5.7.x are very similar, there are some subtle [differences](https://mariadb.com/kb/en/library/function-differences-between-mariadb-103-and-mysql-57/) (notably JSON); in all likelihood this won't be a problem for us, but something to watch out for.

File: README.md [hunk 2]
```diff
@@ -41,6 +41,67 @@ information about using Docker Compose.
 The Compose file included here deploys all services on a custom network, and
 exposes the gateway service at port 8000 on your local machine.
 
+To start up (from the root of the repo, containing ``docker-compose.yml``):
+
+```bash
+$ docker-compose build
```
[the remark this answers is not in the document]
Comment (erickpeirson) [reply README.md]: That looks like an old docker-compose...

[the remark this answers is not in the document]
Comment (erickpeirson) [reply README.md]: @mhl10 Can you verify that your ``docker-compose.yml`` looks like this? https://github.com/cul-it/arxiv-submission-core/blob/task/ARXIVNG-413/docker-compose.yml

File: Dockerfile-metadata (hunk 1 above)
Comment (erickpeirson) [reply Dockerfile-metadata]: Good to know; at this point this project isn't using JSON functions, just the column/dtype, so we should be OK for now.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mhl10) [reply README.md]: Please disregard previous comment. I think I was working from the wrong branch on my home machine. I was able to build successfully on my work machine.

File: core/events/services/classic/bootstrap.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+def _get_locale() -> str:
+    return LOCALES[random.randint(0, len(LOCALES) - 1)]
+
+
+def _epoch(t: datetime) -> int:
+    return int((t - datetime.utcfromtimestamp(0)).total_seconds())
+
+
+LICENSES = [
```
Comment (mhl10) [inline core/events/services/classic/bootstrap.py]: We should consider adding this to arxiv-base in the future.

Comment (erickpeirson) [reply core/events/services/classic/bootstrap.py]: Agreed

**Pull request merged**